src/test/java/com/example/demo/services/BooksServiceImplTest.java

```diff
@@ -16,16 +16,17 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.*;
 
 @RunWith(MockitoJUnitRunner.class)
 @SpringBootTest
 public class BooksServiceImplTest {
     @Mock
     private BookRepository bookRepository;
+    @Mock
+    private BookMainInfo bookMainInfo;
```

Comment:
nie powinno się mockować obiektów które przyjmuje metoda. Nie ma to po prostu sensu. Mockuj @Componenty/@Beany jedynie czyli to co wstrzykujesz przez konstruktor i nic więcej. BookMainInfo powinien być normalnie stworzony przez `new` i uzupełniony tak jak trzeba -> świadomie <- na potrzeby testu

Reply:
done